src/main/java/edu/agh/wfiis/designpatterns/patternsUsageExample/DesignPatternsManager.java

```diff
@@ -59,21 +43,24 @@ public String generateDesignPatternsCourseworkForStudents(){
         return "very difficult coursework";
     }
 
-}
-
-enum PatternName {
-    BUILDER, DECORATOR, STRATEGY;
-}
+    static class PatternHandler {
```

Comment:
Class `PatternHandler` does not meet SRP. In fact, imo, obscure code and makes `DesignPattern` harder to use.

What is more, class is just dumb wrapper, because all of this functionality we are able to get directly from `DesignPattern` object.

Class completely obsolete.